examples/refinement/point_refinement_demo.cc

```diff
@@ -0,0 +1,164 @@
+/**
+ * @file point_refinement_demo.cc
+ * @brief example code for building and pointwise refining of a tensor product
+ *        mesh
+ *
+ * This code generates a tensor-product mesh and performs a user specified
+ * number of steps of pointwise uniform refinement on the cell containing the
+ * point (0.5, 0.5).
+ *
+ * In the end the information about all meshes created in the process
+ * is stored in the form of MATLAB functions.
+ */
+
+#include <functional>
+#include <iostream>
+
+#include <lf/refinement/mesh_hierarchy.h>
+#include <lf/refinement/refutils.h>
+#include "lf/base/base.h"
+#include "lf/mesh/hybrid2dp/hybrid2dp.h"
+#include "lf/mesh/test_utils/check_mesh_completeness.h"
+#include "lf/mesh/test_utils/test_meshes.h"
+#include "lf/mesh/utils/utils.h"
+
+using CodimMeshDataSet_t = std::shared_ptr<lf::mesh::utils::CodimMeshDataSet<bool>>;
+
+CodimMeshDataSet_t
+MarkMesh(const std::shared_ptr<const lf::mesh::Mesh> &mesh_ptr,
+         const Eigen::MatrixXd &point) {
+    // We use CodimMeshDataSet to store if the edges have to be marked
+    CodimMeshDataSet_t marked =
+            lf::mesh::utils::make_CodimMeshDataSet<bool>(mesh_ptr, 1, false);
+
+    // Loop through all cells to check if it contains the point
+    for (const lf::mesh::Entity &cell : mesh_ptr->Entities(0)) {
+        const lf::geometry::Geometry *geom_ptr = cell.Geometry();
+        lf::base::RefEl ref_el = cell.RefEl();
+        const Eigen::MatrixXd &ref_el_coords(ref_el.NodeCoords());
+        const Eigen::MatrixXd glob_coords(geom_ptr->Global(ref_el_coords));
```

Comment:
The same is true for these lines: I think `auto geom_ptr = cell.Geometry();` is much nicer to read...